packages/wp-story-editor/src/api/media.js

```diff
@@ -85,13 +85,35 @@ export function getMediaById(config, mediaId) {
   return apiFetch({ path }).then(getResourceFromAttachment);
 }
 
+/**
+ * Get the optimized variant of a given media resource.
+ *
+ * @param {Object} config Configuration object.
+ * @param {number} mediaId Media ID.
+ * @return {Promise<import('@web-stories-wp/media').Resource>} Media resource if found, null otherwise.
+ */
+export async function getOptimizedMediaById(config, mediaId) {
+  const path = addQueryArgs(`${config.api.media}${mediaId}/`, {
+    context: 'edit',
+    _fields: 'meta.web_stories_optimized_id',
+  });
+
+  const result = await apiFetch({ path });
+
+  if (result?.meta?.web_stories_optimized_id) {
+    return getMediaById(config, result?.meta?.web_stories_optimized_id);
```

Comment:
Does this need any await? 
```suggestion
    const response = await getMediaById(config, result?.meta?.web_stories_optimized_id);
    return response;
```

Reply:
Also, do we need the questions marks here, we know it is set, from the line above.

Reply:
No `await` needed because the Promise is just returned right away and this is an async function.

In short, here `return await Promise` and `return Promise` are identical.